### PR TITLE
Show domain info when signing messages

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/28048

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Sing message ui should show domain field content for v3 and v4 signature types

Desktop: https://github.com/brave/brave-core/pull/16935#issuecomment-1411811281
Android:
image

### CODE REVIEW

File: components/brave_wallet_ui/components/extension/sign-panel/index.tsx [hunk 1]
```diff
@@ -137,6 +137,9 @@ export const SignPanel = (props: Props) => {
     }
   }, [showWarning])
 
+  // TODO(apaymyshev): implement ui for this.
+  console.log(selectedQueueData.domain)
+
```
Comment (supermassive) [inline components/brave_wallet_ui/components/extension/sign-panel/index.tsx:142]: Discussed with @onyb to have ui here similar to 
image

**Review by darkdh — APPROVED** [review]

**Comment by onyb** [comment]
This what the panel looks like after my UI updates in b3cf58e.

|Personal Sign|Sign Typed Data V3| Sign Typed Data V4| Sign Typed Data V4 (⚠️)
-|-|-|-
|||||

**Review by Douglashdaniel — APPROVED** [review]
++ Front-end

**Comment by yrliou** [comment]
Added @SergeyZhukovsky as a reviewer for Android commit.

File: browser/ui/android/strings/android_brave_strings.grd [hunk 2]
```diff
@@ -3365,6 +3365,12 @@ If you don't accept this request, VPN will not reconnect and your internet conne
     <message name="IDS_WALLET_NON_ASCII_CHARACTERS_ASCII" desc="A title to view an ASCII message if Unicode characters detected">
       View message in ASCII encoding
     </message>
+    <message name="IDS_WALLET_SIGN_MESSAGE_DETAILS_DOMAIN_SECTION" desc="A domain section template of sign message details">
+      <ph name="BEGIN_BOLD">&lt;b&gt;</ph>Domain:<ph name="END_BOLD">&lt;/b&gt;</ph><ph name="BREAK">&lt;br&gt;</ph><ph name="ESCAPED_DOMAIN_TEXT">%1$s</ph><ph name="BREAK">&lt;br&gt;</ph>
+    </message>
+    <message name="IDS_WALLET_SIGN_MESSAGE_DETAILS_MESSAGE_SECTION" desc="A message section template of sign message details">
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:3371]: please add ` translateable="false"` for both strings as nothing to translate there.

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.50.3 Chromium: 110.0.5481.52 (Official Build) nightly (64-bit)
-- | --
Revision | `979113183ded4544a2c443aceb1629c430907e52-refs/branch-heads/5481@{#636}`
OS | Linux

- Verified domain info is shown on signing messages for v3/v4 messages

Personal Sign | Sign Type V3 | Sign Type V4
-- | -- | --
image | image | image

-------------------

Verification passed on Oppo Reno 5 with Android 13 running 1.50.3 x64 Nightly build
- Verified domain info is shown on signing messages for v3/v4 messages

https://user-images.githubusercontent.com/17010094/216597758-15694f86-f8b8-4c47-b4dc-b454361a4aba.mp4

**Pull request merged**